### PR TITLE
create a generate jwt functionand do test setup

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
This PR does the initial test setup and adds a utility to generate a jwt token

Test coverage:
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,11 +4,19 @@
...
     "deploy": "wrangler publish",
-    "start": "wrangler dev"
+    "start": "wrangler dev",
+    "test": "jest --coverage ."
   },
   "devDependencies": {
     "@cloudflare/workers-types": "^4.20230419.0",
-    "typescript": "^5.0.4",
+    "@types/jest": "^29.5.3",
+    "jest": "^29.6.1",
```
Comment (iamitprakash) [inline package.json]: can we remove ^ and use stable version

File: package.json [hunk 2]
```diff
@@ -4,11 +4,19 @@
...
+    "ts-jest": "^29.1.1",
+    "ts-node": "^10.9.1",
+    "typescript": "^5.1.6",
     "wrangler": "^3.0.0"
```
Comment (iamitprakash) [inline package.json]: same here

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: src/utils/generateJwt.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import jwt from '@tsndr/cloudflare-worker-jwt';
+import { env } from '../types/global.types';
+
+export const generateJwt = async (env: env) => {
```
Comment (iamitprakash) [inline src/utils/generateJwt.ts:4]: exp claim in the JWT token should be in the future, not in the past. The current time is 2023-07-24 11:41:54 PST, so the exp claim should be set to a value greater than that. For example, you could set it to Math.floor(Date.now() / 1000) + 300

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: src/utils/generateJwt.ts (hunk 3 above)
Comment (RitikJaiswal75) [reply src/utils/generateJwt.ts:4]: We get the server allocated to us only for 10ms. And as per the recent usage we have found this timestamp suits us.

**Review by iamitprakash — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**